src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const ADD_ACTION = 'addProperties';
+  const REMOVE_ACTION = 'removeProperties';
+  const CLEAR_ACTION = 'clear';
+
+  // Array to hold clones of `state` after each action
+  const stateClones = [];
+  const prevClone = { ...state };
```

Comment:
```suggestion
  const prevClone = { ...state };
```
why prev? It's looks like copy ;)

stateCopy sounds better for me